src/rust/ide/src/double_representation/module.rs

```diff
@@ -6,11 +6,192 @@ use crate::double_representation::definition;
+    ///
+    /// Implicitly will trim the whitespace from input name to ease usage with `import` macro body.
+    fn from_qualified_name(name:impl AsRef<str>) -> Self {
+        let name     = name.as_ref().trim();
+        let segments = name.split(ast::opr::predefined::ACCESS).map(Into::into).collect();
+        ImportInfo { target: segments }
+    }
+
+    fn qualified_name(&self) -> QualifiedName {
+        QualifiedName(self.target.join(ast::opr::predefined::ACCESS))
+    }
+
+    fn from_ast(ast:&Ast) -> Option<Self> {
+        let macro_match = known::Match::try_from(ast).ok()?;
+        Self::from_match(macro_match)
+    }
+
+    fn from_match(ast:known::Match) -> Option<Self> {
+        ast::macros::is_match_import(&ast).then_with(|| {
+            ImportInfo::from_qualified_name(ast.segs.head.body.repr())
+        })
+    }
+}
+
+impl Display for ImportInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{} {}",ast::macros::IMPORT_KEYWORD,self.qualified_name())
+    }
+}
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+#[derive(Clone,Debug,Fail)]
+#[fail(display="Import `{}` was not found in the module.",_0)]
+#[allow(missing_docs)]
+pub struct ImportNotFound(pub ImportInfo);
+
+#[derive(Clone,Copy,Debug,Fail)]
+#[fail(display="Line index is out of bounds.")]
+#[allow(missing_docs)]
+pub struct LineIndexOutOfBounds;
+
+
+
+// ============
+// === Info ===
+// ============
+
+/// Wrapper allowing getting information about the module and updating it.
+#[derive(Clone,Debug)]
+pub struct Info {
+    #[allow(missing_docs)]
+    pub ast:known::Module,
+}
+
+impl Info {
+    /// Iterate over all lines in module that contain an import declaration.
+    pub fn enumerate_imports<'a>(&'a self) -> impl Iterator<Item=(ModuleCrumb, ImportInfo)> + 'a {
+        let children = self.ast.shape().enumerate();
+        children.filter_map(|(crumb,ast)| Some((crumb,ImportInfo::from_ast(ast)?)))
+    }
+
+    /// Add a new line to the module's block.
+    ///
+    /// Note that indices are the "module line" indices, which usually are quite different from text
+    /// API line indices (because nested blocks doesn't count as separate "module lines").
+    pub fn add_line(&mut self, index:usize, ast:Option<Ast>) {
+        let line = BlockLine::new(ast);
+        self.ast.update_shape(|shape| shape.lines.insert(index,line))
+    }
+
+    /// Remove line with given index.
+    ///
+    /// Returns removed line. Fails if the index is out of bounds.
+    pub fn remove_line(&mut self, index:usize) -> FallibleResult<BlockLine<Option<Ast>>> {
+        self.ast.update_shape(|shape| {
+            shape.lines.try_remove(index).ok_or(LineIndexOutOfBounds.into())
+        })
+    }
+
+    /// Get all import declarations in the module.
+    pub fn imports(&self) -> Vec<ImportInfo> {
+        self.enumerate_imports().map(|(_,import)| import).collect()
+    }
```

Comment:
For me it's a little more than "get" because we're generating Vec here. Adjust docs, or return just iterator (and the user can decide if he wants to generate Vec from it).